doc/helm/dev/developer_guide.md

```diff
@@ -0,0 +1,455 @@
+[install Helm][helm_install] and test your charts using the `helm` command line
+tool.
+
+**NOTE:** Installing Helm's Tiller component in your cluster is not required.
+
+### Testing a Helm chart locally
+
+Sometimes it is beneficial for a developer to run the Helm chart installation
+from their local machine as opposed to running/rebuilding the operator each
+time. To do this, initialize a new project:
+
+```sh
+$ operator-sdk new --type helm --kind Foo --api-version foo.example.com/v1alpha1 foo-operator
+INFO[0000] Creating new Helm operator 'foo-operator'.
+INFO[0000] Create build/Dockerfile
+INFO[0000] Create watches.yaml
+INFO[0000] Create deploy/service_account.yaml
+INFO[0000] Create deploy/role.yaml
+INFO[0000] Create deploy/role_binding.yaml
+INFO[0000] Create deploy/operator.yaml
+INFO[0000] Create deploy/crds/foo_v1alpha1_foo_crd.yaml
+INFO[0000] Create deploy/crds/foo_v1alpha1_foo_cr.yaml
+INFO[0000] Create helm-charts/foo/
+INFO[0000] Run git init ...
+Initialized empty Git repository in /home/joe/go/src/github.com/operator-framework/foo-operator/.git/
+INFO[0000] Run git init done
+INFO[0000] Project creation complete.
+
+$ cd foo-operator
+```
+
+For this example we will use the default Nginx helm chart scaffolded by
+`operator-sdk new`. Without making any changes, we can see what the default
+release manifests are:
+
+```sh
+$ helm template --name test-release helm-charts/foo
+---
+# Source: foo/templates/service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: test-release-foo
+  labels:
+    app.kubernetes.io/name: foo
+    helm.sh/chart: foo-0.1.0
+    app.kubernetes.io/instance: test-release
+    app.kubernetes.io/managed-by: Tiller
+spec:
+  type: ClusterIP
+  ports:
+    - port: 80
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app.kubernetes.io/name: foo
+    app.kubernetes.io/instance: test-release
+
+---
+# Source: foo/templates/deployment.yaml
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: test-release-foo
+  labels:
+    app.kubernetes.io/name: foo
+    helm.sh/chart: foo-0.1.0
+    app.kubernetes.io/instance: test-release
+    app.kubernetes.io/managed-by: Tiller
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: foo
+      app.kubernetes.io/instance: test-release
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: foo
+        app.kubernetes.io/instance: test-release
+    spec:
+      containers:
+        - name: foo
+          image: "nginx:stable"
+          imagePullPolicy: IfNotPresent
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
+          resources:
+            {}
+
+
+---
+# Source: foo/templates/ingress.yaml
```

Comment:
What is this source for?

Reply:
When Helm templates a release manifest, it puts the name of every template YAML file in the rendered manifest, even if individual templates render to an empty string, which is what happens in this case since the ingress is disabled by default.